packages/garbo/src/resources/candyMap.ts

```diff
@@ -0,0 +1,123 @@
+import { AcquireItem, Outfit } from "grimoire-kolmafia";
+import {
+  $familiar,
+  $item,
+  get,
+  getSaleValue,
+  have,
+  sum,
+} from "libram";
+import {
+  canEquip,
+  getOutfits,
+  mallPrice,
+  outfit,
+  outfitPieces,
+  outfitTreats,
+  print,
+  runChoice,
+  toItem,
+  use,
+  visitUrl,
+} from "kolmafia";
+import { GarboTask } from "../tasks/engine";
+import { garboValue } from "../garboValue";
+
+const HOUSE_NUMBERS = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
+
+let blockHtml = "";
+
+function getBlockHtml(): string {
+  blockHtml ||= visitUrl("place.php?whichplace=town&action=town_trickortreat");
+  return blockHtml;
+}
+
+function treatValue(outfit: string): number {
+  return sum(
+    Object.entries(outfitTreats(outfit)),
+    ([candyName, probability]) => probability * garboValue(toItem(candyName))
+  );
+}
+
+export function getTreatOutfit(): string {
+  const outfits = getOutfits();
+  const outfitsWithValues = outfits.map((outfit) => ({
+    outfit,
+    relativeValue: treatValue(outfit),
+}));
+
+const outfitWithHighestValue = outfitsWithValues.reduce((prev, current) => {
+  return prev.relativeValue > current.relativeValue ? prev : current;
+});
+
+  return outfitWithHighestValue.outfit;
+ //ChatGPT made this, I have no idea if it works!
+}
```

Comment:
```suggestion
export function getTreatOutfit(): string {
  const availableOutfits = getOutfits().filter((name) =>
    outfitPieces(name).every((piece) => canEquip(piece))
  );
  if (!availableOutfits.length) {
    print("You don't seem to actually have any outfits available, my friend!");
  }
  return maxBy(availableOutfits, treatValue);
}
```

Reply:
`maxBy` is more or less equivalent to this reduce, but a little cleaner to read here. We need to check the array length because reducing an empty array can be dangerous. We need to filter the result of `getOutfits` to only include outfits we can equip